src/Microsoft.TemplateEngine.Abstractions/GlobalSettings/IGlobalSettings.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.TemplateEngine.Abstractions.GlobalSettings
+{
+    /// <summary>
+    /// Interface that represents loading/storing data into settings.json file.
+    /// That is shared between multiple different hosts of TemplateEngine.
+    /// </summary>
+    public interface IGlobalSettings
+    {
+        /// <summary>
+        /// Triggered every time when settings change.
+        /// </summary>
+        event Action SettingsChanged;
+
+        /// <summary>
+        /// Returns uncached list of installer template packages.
+        /// </summary>
+        Task<IReadOnlyList<TemplatePackageData>> GetInstalledTemplatesPackagesAsync(CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Stores list of installer template packages.
+        /// </summary>
+        Task SetInstalledTemplatesPackagesAsync(IReadOnlyList<TemplatePackageData> packages, CancellationToken cancellationToken);
```

Comment:
As we discussed on call, since this was made private since comment was posted, I'm marking this as resolved...
This method is pretty much Load/Save set of packages...